core/src/test/kotlin/com/kin/ecosystem/core/data/order/ExternalEarnOrderCallTest.kt

```diff
@@ -0,0 +1,107 @@
+package com.kin.ecosystem.core.data.order
+
+import com.kin.ecosystem.common.KinCallback
+import com.kin.ecosystem.common.Observer
+import com.kin.ecosystem.core.bi.EventLogger
+import com.kin.ecosystem.core.bi.events.EarnOrderCompletionSubmitted
+import com.kin.ecosystem.core.bi.events.EarnOrderCreationReceived
+import com.kin.ecosystem.core.data.blockchain.BlockchainSource
+import com.kin.ecosystem.core.data.blockchain.Payment
+import com.kin.ecosystem.core.network.model.BlockchainData
+import com.kin.ecosystem.core.network.model.JWTBodyPaymentConfirmationResult
+import com.kin.ecosystem.core.network.model.OpenOrder
+import com.kin.ecosystem.core.network.model.Order
+import com.nhaarman.mockitokotlin2.*
+import kin.ecosystem.test.base.BaseTestClass
+import kin.sdk.migration.common.KinSdkVersion
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import java.math.BigDecimal
+
+@Config(manifest = Config.NONE)
+@RunWith(RobolectricTestRunner::class)
+class ExternalEarnOrderCallTest : BaseTestClass() {
+
+	private val orderDataSource: OrderDataSource = mock {
+		on { createExternalOrderSync(any()) } doAnswer { openOrder}
+	}
+	private val blockchainSource: BlockchainSource = mock {
+		on { blockchainVersion } doAnswer { KinSdkVersion.OLD_KIN_SDK }
+	}
+	private val eventLogger: EventLogger = mock()
+	private val externalOrderCallbacks: CreateExternalOrderCall.ExternalOrderCallbacks = mock()
+
+	private val openOrder: OpenOrder = mock {
+		on { offerId } doAnswer  { offerId }
+		on { id } doAnswer { orderId }
+		on { amount } doAnswer { 10 }
+		on { blockchainData } doAnswer { blockchainData }
+	}
+
+	private val blockchainData: BlockchainData = mock {
+		on { recipientAddress } doAnswer { "some_address" }
+	}
+
+	private val order: Order = mock {
+		on { orderId } doAnswer { orderId }
+		on { offerId } doAnswer { offerId }
+		on { result } doAnswer { jwtBodyPaymentConfirmationResult }
+	}
+
+	private val payment: Payment = mock {
```

Comment:
why do you need to mock model pojos? (`Order`,`BlockchainData`)
I think it's better to simply instantiate them with the right values